### PR TITLE
Restart

### PR DESCRIPTION
**Pull request by nikhar-abbas**
## Description and Purpose
This adds restart capabilities to ROSCO. 

TODO:
- [ ] Set up automated ROSCO_Type generation and checkpoint file writing
- [ ] Test and compare with previous results

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

### CODE REVIEW

File: ROSCO/src/Functions.f90 [hunk 1]
```diff
@@ -627,7 +623,7 @@ SUBROUTINE Debug(LocalVar, CntrPar, DebugVar, avrSWAP, RootName, size_avcOUTNAME
             END IF
             
             IF (CntrPar%LoggingLevel > 1) THEN 
-                OPEN(unit=UnDb2, FILE=RootName(1:size_avcOUTNAME-5)//'RO.dbg2')
+                OPEN(unit=UnDb2, FILE='DEBUG2.dbg')
```
Comment (dzalkind) [inline ROSCO/src/Functions.f90:626]: Can you revert this change back to what it was?

**Comment by nikhar-abbas** [comment]
Closing this, opening a new one with a new branch
